dias/analyzers/sensitivity_analyzer.py

```diff
@@ -0,0 +1,273 @@
+                
+                dset = handler.create_dataset('count', data=counter.astype(np.int))
+                dset.attrs['axis'] = np.array(['freq', 'pol', 'time'], dtype='S')
+                
+                handler.attrs['instrument_name'] =self.correlator
+                handler.attrs['collection_server'] = subprocess.check_output(["hostname"]).strip()
+                handler.attrs['system_user'] = subprocess.check_output(["id", "-u", "-n"]).strip()
+                handler.attrs['git_version_tag'] = subprocess.check_output(["git", "-C", 
+                                                                            os.path.dirname(__file__), 
+                                                                            "describe", "--always"]).strip()
+
+    def get_baselines(self, indexmap, inputmap):
+        
+        prod = defaultdict(list)
+        prodmap = defaultdict(list)
+        dist = defaultdict(list)
+        conj = defaultdict(list)
+        cyl = defaultdict(list)
+        scale = defaultdict(list)
+
+        feedpos = tools.get_feed_positions(inputmap)
+        
+        for pp, (this_prod, this_conj) in enumerate(indexmap['stack']):
+
+            if this_conj:
+                bb, aa = indexmap['prod'][this_prod]
+            else:
+                aa, bb = indexmap['prod'][this_prod]
+            
+            inp_aa = inputmap[aa]
+            inp_bb = inputmap[bb]
+            
+            if not tools.is_chime(inp_aa) or not tools.is_chime(inp_bb):
+                continue
+                
+            if not self.include_auto and (aa == bb):
+                continue
+
+            if not self.include_intracyl and (inp_aa.cyl == inp_bb.cyl):
+                continue
+
+            this_dist = list(feedpos[aa, :] - feedpos[bb, :])
+
+            if tools.is_array_x(inp_aa) and tools.is_array_x(inp_bb):
+                key = 'XX'
+
+            elif tools.is_array_y(inp_aa) and tools.is_array_y(inp_bb):
+                key = 'YY'
+
+            elif not self.include_crosspol:
+                continue
+                
+            elif tools.is_array_x(inp_aa) and tools.is_array_y(inp_bb):
+                key = 'XY'
+
+            elif tools.is_array_y(inp_aa) and tools.is_array_x(inp_bb):
+                key = 'YX'
+
+            else:
+                raise RuntimeError("CHIME feeds not polarized.")
+
+            this_cyl = '%s%s' % (self.get_cyl(inp_aa.cyl), self.get_cyl(inp_bb.cyl))
+            if self.sep_cyl:
+                key = key + '-' + this_cyl
+
+            prod[key].append(pp)
+            prodmap[key].append((aa, bb))
+            conj[key].append(this_conj)
+            dist[key].append(this_dist)
+            cyl[key].append(this_cyl)
+
+            if aa == bb:
+                scale[key].append( 0.5 )
+            else:
+                scale[key].append( 1.0 )
+
+
+        for key in prod.keys():
```

Comment:
Shy don't you do this in the loop above?

Reply:
Are you referring to `for key in prod.keys():` loop or `get_baselines` function itself?

Reply:
This inline comment is about the line it's attached to, so `for key in prod.keys():`. I didn't see a reason why you wouldn't just do it in the loop above (for performance).

Reply:
Given that it's being called once for every run (i.e. the get_baseline function is above the files loop as you suggested), the declaration to arrays should be not be a large overhead.